generate_capacitor_radial_tht.py

```diff
@@ -0,0 +1,371 @@
+"""
+Generate THT polarized radial electrolytic capacitors (CAPPRD).
+"""
+from os import makedirs, path
+from uuid import uuid4
+
+from typing import Optional
+
+from common import init_cache, now, save_cache
+from entities.common import (
+    Align, Angle, Author, Category, Circle, Created, Deprecated, Description, Diameter, Fill, GrabArea, Height,
+    Keywords, Layer, Name, Polygon, Position, Rotation, Value, Version, Vertex, Width
+)
+from entities.component import SignalUUID
+from entities.device import ComponentPad, ComponentUUID, Device, PackageUUID
+from entities.package import (
+    AutoRotate, Drill, Footprint, FootprintPad, LetterSpacing, LineSpacing, Mirror, Package, PackagePad, Shape, Side,
+    Size, StrokeText, StrokeWidth
+)
+
+generator = 'librepcb-parts-generator (generate_capacitor_radial_tht.py)'
+
+# Lookup table to get the drill diameter from a given lead diameter.
+LEAD_WIDTH_TO_DRILL = {
+    0.4: 0.5,
+    0.45: 0.6,
+    0.5: 0.7,
+    0.6: 0.8,
+    0.8: 1.0,
+}
+
+# Initialize UUID cache
+uuid_cache_file = 'uuid_cache_capacitors_radial_tht.csv'
+uuid_cache = init_cache(uuid_cache_file)
+
+
+def uuid(category: str, full_name: str, identifier: str) -> str:
+    """
+    Return a uuid for the specified element.
+
+    Params:
+        category:
+            For example 'cmp' or 'pkg'.
+        full_name:
+            For example "SOIC127P762X120-16".
+        identifier:
+            For example 'pad-1' or 'pin-13'.
+    """
+    key = '{}-{}-{}'.format(category, full_name, identifier).lower().replace(' ', '~')
+    if key not in uuid_cache:
+        uuid_cache[key] = str(uuid4())
+    return uuid_cache[key]
+
+
+def get_variant(
+    diameter: float,
+    height: float,
+    pitch: float,
+    lead_width: float,
+) -> str:
+    return 'D{}-H{}-P{}-W{}'.format(diameter, height, pitch, lead_width)
+
+
+def generate_pkg(
+    dirpath: str,
+    diameter: float,
+    height: float,
+    pitch: float,
+    lead_width: float,
+    author: str,
+    version: str,
+    create_date: Optional[str],
+) -> None:
+    # Name according IPC-7351 "Capacitor, Polarized Radial Diameter":
+    # CAPPRD + Lead Spacing + W Lead Width + D Body Diameter + H Body Height
+    name = 'CAPPRD{:.0f}W{:.0f}D{:.0f}H{:.0f}'.format(
+        pitch * 100, lead_width * 100, diameter * 100, height * 100)
```

Comment:
For the numbers we actually have a [helper function](https://github.com/LibrePCB/librepcb-parts-generator/blob/master/common.py#L58-L64). (The trick with `:.0f` looks nice though, maybe the helper function can use that too if it covers all cases.)

Reply:
Ah, I didn't know that function -> adjusted. There is no difference in the generated files, so it seems to work properly :)